arm_robots/src/arm_robots/hdt_michigan.py

```diff
@@ -30,20 +29,29 @@ def __init__(self, robot_namespace: str):
         self.latest_cmd = JointState()
         self.command_rate = rospy.Rate(100)
         self.ready = 0
+        self.has_started_command_thread = False
 
     def __del__(self):
         self.disconnect()
 
     def connect(self):
         super().connect()
-        self.command_thread.start()
+        if not self.has_started_command_thread:
+            rospy.loginfo('Starting val command thread')
+            self.has_started_command_thread = True
+            self.command_thread.start()
 
     def disconnect(self):
+        super().disconnect()
+
         self.should_disconnect = True
         if self.command_thread.is_alive():
             self.command_thread.join()
 
     def command_thread_func(self):
+        if rospy.get_param("use_sim_time", False):
```

Comment:
Why put this check here instead of in `def connect` above?

Reply:
good point, that's a bit simpler